tutorials/generative/realism_diversity_metrics/realism_diversity_metrics.py

```diff
@@ -0,0 +1,356 @@
+    image = subtract_mean(image)
+
+    # Get model outputs
+    with torch.no_grad():
+        feature_image = radnet.forward(image)
+        # flattens the image spatially
+        feature_image = spatial_average(feature_image, keepdim=False)
+
+    return feature_image
+
+
+# -
+
+# ## Setup data directory
+#
+# You can specify a directory with the MONAI_DATA_DIRECTORY environment variable.
+# This allows you to save results and reuse downloads.
+#
+# If not specified a temporary directory will be used.
+
+directory = os.environ.get("MONAI_DATA_DIRECTORY")
+root_dir = tempfile.mkdtemp() if directory is None else directory
+print(root_dir)
+
+# ## Set deterministic training for reproducibility
+
+set_determinism(5)
+
+# ## Define the models
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+print(f"Using {device}")
+
+autoencoderkl = AutoencoderKL(
+    spatial_dims=2,
+    in_channels=1,
+    out_channels=1,
+    latent_channels=3,
+    num_channels=[64, 128, 128],
+    num_res_blocks=1,
+    norm_num_groups=32,
+    attention_levels=(False, False, True),
+)
+autoencoderkl = autoencoderkl.to(device)
+
+# +
+unet = DiffusionModelUNet(
+    spatial_dims=2,
+    in_channels=1,
+    out_channels=1,
+    num_res_blocks=(1, 1, 1),
+    num_channels=(64, 128, 128),
+    attention_levels=(False, True, True),
+    num_head_channels=128
+)
+
+scheduler = DDIMScheduler(num_train_timesteps=1000, beta_schedule="linear", beta_start=0.0015, beta_end=0.0195)
+
+inferer = DiffusionInferer(scheduler)
+
+discriminator = PatchDiscriminator(
+    spatial_dims=2,
+    num_layers_d=3,
+    num_channels=32,
+    in_channels=1,
+    out_channels=1,
+    kernel_size=4,
+    activation=(Act.LEAKYRELU, {"negative_slope": 0.2}),
+    norm="BATCH",
+    bias=False,
+    padding=1,
+)
+discriminator.to(device)
+unet = unet.to(device)
+# -
+
+# ## Load pre-trained model
+
+# Here we will use a pre-trained version of the DDPM downloaded from torch. However, users can also use a local model and evaluate its metrics if they want.
+
+use_pre_trained = True
+
+if use_pre_trained:
+    unet = torch.hub.load("marksgraham/pretrained_generative_models:v0.2", model="ddpm_2d", verbose=True)
+    unet = unet.to(device)
+else:
+    model_path = Path.cwd() / Path("tutorials/generative/2d_ldm/best_aeutoencoderkl.pth")
+    autoencoderkl.load_state_dict(torch.load(str(model_path)))
+    model_path = Path.cwd() / Path("tutorials/generative/2d_ldm/best_unet.pth")
+    unet.load_state_dict(torch.load(str(model_path)))
+
+# ## Get the real images
+
+# Simialry to the 2D LDM tutorial here we will use the MedNISTDataset, which contains images from different body parts. For easiness, here we will use only the `Hand` class. The first part of the code will get the real images from the MedNISTDataset and apply some transformation to scale the intensity of the image. Because we are evaluating the performance of the trained network, we will only use the validation split.
```

Comment:
Consider using
**Similar to the 2D LDM tutorial, we** will use the MedNISTDataset, which contains images from different body parts. For easiness, here we will use only the `Hand` class. The first part of the code will get the real images from the MedNISTDataset and apply some **transformations** to scale the intensity of the image. Because we are evaluating the performance of the trained network, we will only use the validation split.